lib/ansible/modules/cloud/vmware/vmware_host_sriov.py

```diff
@@ -0,0 +1,332 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    "metadata_version": "1.1",
+    "status": ["preview"],
+    "supported_by": "community",
+}
+
+DOCUMENTATION = r"""
+---
+module: vmware_host_sriov
+short_description: Manage SR-IOV settings on host
+description:
+- This module can be used to configure, enable/disable SR-IOV functions an ESXi host.
+- module didn't reboot host after changes, but put in output "rebootRequired" key.
+- User can specify an ESXi hostname or Cluster name. In case of cluster name, all ESXi hosts are updated.
+version_added: 2.9
+author:
+- Viktor Tsymbalyuk (@victron)
+notes:
+- Tested on vSphere 6.0
+requirements:
+- python >= 2.6
+- PyVmomi
+options:
+  esxi_hostname:
+    description:
+    - Name of the host system to work with.
+    - This parameter is required if C(cluster_name) is not specified.
+    type: str
+  cluster_name:
+    description:
+    - Name of the cluster from which all host systems will be used.
+    - This parameter is required if C(esxi_hostname) is not specified.
+    type: str
+  vmnic:
+    description:
+    - interface name, like vmnic0
+    type: str
```

Comment:
Can this be a list of NICs ?

Reply:
I thought about it. But in reason that it's my first experience, I prefer leave it as is.  A little bit late, I'll redo it on [str,  list].

Reply:
I'll do it in next version. Because may be it's better use dict instead, for example:
```
[{ "vmnic": "vmnic0",  "num_virt_func": 8},
{ "vmnic": "vmnic1",  "num_virt_func": 0}]